### PR TITLE
CBOR2 preconfigured converter

### PR DESCRIPTION
**Pull request by brendan-simon-indt**
Add a preconfigured converter for the python `cbor2` package, for CBOR encoding/decoding.

This was modeled after the `msgpack` converter, as they are very similar.

Hope this is acceptable.  Let me know if there is anything else required.

### CODE REVIEW

**Review by Tinche — APPROVED** [review]
This looks pretty great, thanks!

Let's see if the tests pass and we can start merging.

**Comment by Tinche** [comment]
Oh yeah, and please add a HISTORY entry

File: src/cattr/preconf/cbor2.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""Preconfigured converters for cbor2."""
```
[the remark this answers is not in the document]
Comment (brendan-simon-indt) [reply src/cattr/preconf/cbor2.py]: Done

**Comment by brendan-simon-indt** [comment]
Updates done as per PR comments.  Do I do another PR or is just pushing the changes enough?

**Comment by Tinche** [comment]
Just push the changes, no need to complicate matters. Thanks!

**Comment by brendan-simon-indt** [comment]
Do any of the failures above require additional work by me?

**Comment by Tinche** [comment]
Yep, looks like we need to add the cbor2 library to the dev dependencies here: https://github.com/brendan-simon-indt/cattrs/blob/cf8580a5098390aa2ede14f7e49ec1ff80474c24/pyproject.toml#L29, so it gets installed for tests.

You can use `poetry add cbor@latest --group dev` to do so with Poetry, and then run `poetry update` to generate the lockfile.

**Comment by Tinche** [comment]
Looks like there's a linting problem at tests/test_preconf.py, I can take care of that though. Thanks!

**Pull request merged**